web/src/utils/date/index.ts

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  differenceInCalendarDays,
+  formatDistanceToNowStrict,
+  isPast,
+} from 'date-fns';
+
+/**
+ * Formats a Date object into a formatted deadline string.
+ * @param date The date of the deadline
+ */
+export const formatDeadlineFromNowText = (date: Date): string => {
+  const now = new Date();
+  const distanceInDays = formatDistanceToNowStrict(date, {unit: 'day'});
+  if (isPast(date)) {
+    return `Ended ${distanceInDays} ago`;
+  }
+  return differenceInCalendarDays(date, now) > 0
+    ? `${distanceInDays} left`
+    : 'Last day!';
```

Comment:
I find it a bit weird that we have `if (isPast(date))` and then another check for the return statement below that. Essentially, what you are checking is if deadline is in the past, in the future or today exactly. I think it's easier to follow the logic if we just combine these two checks like this:
```suggestion
  return isPast(date) ?
    `Ended ${distanceInDays} ago` :
    differenceInCalendarDays(date, now) > 0 ?
      `${distanceInDays} left` :
      'Last day!';
```
or the equivalent if-else statement is also find if you think that is better.

Reply:
Haha personally I'm not a fan of chaining ternery operators, I feel that it makes code less readable. Refer to https://rules.sonarsource.com/javascript/RSPEC-3358, and also https://eslint.org/docs/rules/no-nested-ternary

My thoughts when writing this was having `if (isPast(date))` as an "early termination", and then dealing with those that have not past separately, so I threw them into a single ternery operator.